lib/x11regressiontest.pm

```diff
@@ -454,14 +454,49 @@ sub post_fail_hook() {
 }
 
 sub start_firefox() {
+    my ($self) = @_;
     mouse_hide(1);
 
     x11_start_program 'xterm';
     # Clean and Start Firefox
     type_string "killall -9 firefox;rm -rf .moz* .config/iced* .cache/iced* .local/share/gnome-shell/extensions/*; firefox > firefox.log 2>&1 &\n";
+    $self->firefox_check_default;
+    $self->firefox_check_popups;
     assert_screen 'firefox-launch', 90;
 }
 
+sub firefox_check_default {
+    # Set firefox as default browser if asked
+    assert_screen [qw(firefox_default_browser firefox_trackinfo firefox_readerview_window firefox_clean)], 120;
```

Comment:
Ok, the maybe we can move this set of expected pop-ups into variable, as probably has to be consistent with set of tags on line 482. But that's only suggestion.